app/lib/irs1040_schedule_eic_pdf.rb

```diff
@@ -25,17 +25,29 @@ def dependents_info(dependent_nodes)
     answers = {}
     dependent_nodes.each_with_index do |dependent, index|
       answers["ChildFirstAndLastName#{index + 1}"] = "#{dependent.at("PersonFirstNm").text} #{dependent.at("PersonLastNm").text}"
-        answers["QualifyingChildSSN#{index + 1}"] = dependent.at("QualifyingChildSSN").text
-        answers["ChildBirthYr#{index + 1}[0]"] = dependent.at("ChildBirthYr").text[0]
-        answers["ChildBirthYr#{index + 1}[1]"] = dependent.at("ChildBirthYr").text[1]
-        answers["ChildBirthYr#{index + 1}[2]"] = dependent.at("ChildBirthYr").text[2]
-        answers["ChildBirthYr#{index + 1}[3]"] = dependent.at("ChildBirthYr").text[3]
-        answers["ChildIsAStudentUnder24IndYes#{index + 1}"] =  xml_check_to_bool(dependent.at("ChildIsAStudentUnder24Ind")) ? "Yes" : nil
-        answers["ChildIsAStudentUnder24IndNo#{index + 1}"] = !xml_check_to_bool(dependent.at("ChildIsAStudentUnder24Ind")) ? "Yes" : nil
-        answers["ChildPermanentlyDisabledIndYes#{index + 1}"] = xml_check_to_bool(dependent.at("ChildPermanentlyDisabledInd")) ? "Yes" : nil
-        answers["ChildPermanentlyDisabledIndNo#{index + 1}"] = !xml_check_to_bool(dependent.at("ChildPermanentlyDisabledInd")) ? "Yes" : nil
-        answers["ChildRelationshipCd#{index + 1}"] = dependent.at("ChildRelationshipCd").text
-        answers["MonthsChildLivedWithYouCnt#{index + 1}"] = dependent.at("MonthsChildLivedWithYouCnt").text
+      answers["QualifyingChildSSN#{index + 1}"] = dependent.at("QualifyingChildSSN").text
+      answers["ChildBirthYr#{index + 1}[0]"] = dependent.at("ChildBirthYr").text[0]
+      answers["ChildBirthYr#{index + 1}[1]"] = dependent.at("ChildBirthYr").text[1]
+      answers["ChildBirthYr#{index + 1}[2]"] = dependent.at("ChildBirthYr").text[2]
+      answers["ChildBirthYr#{index + 1}[3]"] = dependent.at("ChildBirthYr").text[3]
+      case xml_boolean_type_value(dependent.at("ChildIsAStudentUnder24Ind"))
+      when true
+        answers["ChildIsAStudentUnder24IndYes#{index + 1}"] = "1"
+        answers["ChildIsAStudentUnder24IndNo#{index + 1}"] = nil
+      when false
+        answers["ChildIsAStudentUnder24IndYes#{index + 1}"] = nil
+        answers["ChildIsAStudentUnder24IndNo#{index + 1}"] = "2"
+      end
+      case xml_boolean_type_value(dependent.at("ChildPermanentlyDisabledInd"))
```

Comment:
The reason for the `case` stuff is that we need to allow ourselves to let `nil` pass through as neither `true` nor `false`.